config/hotreloadable.go

```diff
@@ -1,14 +1,56 @@
 package config
 
-import "time"
+import (
+	"sync"
+	"time"
+)
 
 // RegisterIntConfigVariable registers int config variable
+// Deprecated: use RegisterIntVar or RegisterAtomicIntVar instead
 func RegisterIntConfigVariable(defaultValue int, ptr *int, isHotReloadable bool, valueScale int, keys ...string) {
 	Default.RegisterIntConfigVariable(defaultValue, ptr, isHotReloadable, valueScale, keys...)
 }
 
+// RegisterIntVar registers a not hot-reloadable int config variable
+func RegisterIntVar(defaultValue int, ptr *int, valueScale int, keys ...string) {
```

Comment:
For the new non-reloadable api, instead of passing a pointer to a value for the function to fill, it would be more straightforward for the function to return the config value itself.